src/plugins/ffmemless/plugin.c

```diff
@@ -492,10 +563,35 @@ static int ffm_play(struct ffm_effect_data *data, int play)
 		N_DEBUG (LOG_CAT "Stopping playback");
 	}
 
+#ifdef CACHE_EFFECTS
+	int result = TRUE;
+
+	if (play) {
+		data->cached_effect.id = -1;
+		if (ffmemless_upload_effect(&data->cached_effect, ffm.dev_file)) {
+			N_DEBUG (LOG_CAT "%d effect re-load failed", data->id);
+			result = FALSE;
+		} else {
+			data->id = data->cached_effect.id;
+		}
+	}
+
 	if (ffmemless_play(data->id, ffm.dev_file, play))
-		return FALSE;
+		result = FALSE;
+
+	if (!play) {
+		if (ffmemless_erase_effect(data->cached_effect.id, ffm.dev_file))
```

Comment:
very important to erase the effect at stop, since the device then runs out of 'space'